spec/acceptance/mysql_backup_spec.rb

```diff
@@ -29,55 +27,48 @@ class { 'mysql::server::backup':
         }
     MANIFEST
     it 'when configuring mysql backups' do
-      idempotent_apply(default, pp, {})
+      idempotent_apply(pp)
     end
   end
 
-  describe 'mysqlbackup.sh' do
+  describe 'mysqlbackup.sh', if: Gem::Version.new(mysql_version) < Gem::Version.new('5.7.0') do
     before(:all) do
       pre_run
     end
 
     it 'runs mysqlbackup.sh with no errors' do
-      unless version_is_greater_than('5.7.0')
-        shell('/usr/local/sbin/mysqlbackup.sh') do |r|
-          expect(r.stderr).to eq('')
-        end
+      run_shell('/usr/local/sbin/mysqlbackup.sh') do |r|
```

Comment:
Does the module create the file `mysqlbackup.sh`? It seems like this command `run_shell('/usr/local/sbin/mysqlbackup.sh')` could be moved into a `before(:each)` block, and then the `it`'s on 39 and 52 could be removed in favor of simply the two `it`'s on 45 and 58.
This would have the additional advantage of more properly modeling the relationship that needs to occur between the `run_shell` command that needs to run before each test, and the test itself. Currently the `run_shell` is in an it that is not specifically related to the `it` immediately preceding. The `it` on 39 for example seems like it's just test setup for the it `it` on 45, but that relationship is not currently modeled by the way the file is arranged.
I have no expectation that the current arrangement would fail any time soon, so I'm not proposing to hold up merge for this suggestion.